### PR TITLE
WIP: gh-106593: PyResource API [DO NOT MERGE]

### PR DESCRIPTION
**Pull request by vstinner**
* Add PyUnicode_AsUTF8Resource()
* Add PyBytes_AsStringResource()
* Add PySequence_AsObjectArray()
* Add Include/pyresource.h
* Add PyResource_Release() to the stable ABI
* compute_abstract_methods(): Replace PySequence_Fast() with PySequence_AsObjectArray()

* Issue: gh-106593

----
:books: Documentation preview :books:: https://cpython-previews--106596.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
Proof-of-Concept PR for issue #106592 PyResource API and issue #106593 PyObject_AsObjectArray() API.

If these issues design is accepted, I will write a clean PR with doc and tests.

File: Misc/stable_abi.toml [hunk 1]
```diff
@@ -2432,3 +2432,5 @@
     added = '3.13'
 [function.PyWeakref_GetRef]
     added = '3.13'
+[function.PyResource_Release]
+    added = '3.13'
```
Comment (encukou) [inline Misc/stable_abi.toml:2451]: The `PyResource` struct should be added here as well.

Comment (vstinner) [reply Misc/stable_abi.toml:2451]: Oh right: fixed

**Comment by vstinner** [comment]
I close my PR: https://github.com/python/cpython/issues/106593#issuecomment-1694131973

**Pull request closed without merging**